src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/BaseComponent.cs

```diff
@@ -0,0 +1,186 @@
+            /// </summary>
+            Frame = Interop.CBApplication.NativeComponentType.Frame,
+
+            /// <summary>
+            /// The service component.
+            /// </summary>
+            Service = Interop.CBApplication.NativeComponentType.Service
+        }
+
+        /// <summary>
+        /// Occurs when the system memory is low.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LowMemoryEventArgs> LowMemory;
+
+        /// <summary>
+        /// Occurs when the system battery is low.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LowBatteryEventArgs> LowBattery;
+
+        /// <summary>
+        /// Occurs when the system language is chagned.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LocaleChangedEventArgs> LocaleChanged;
+
+        /// <summary>
+        /// Occurs when the region format is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<RegionFormatChangedEventArgs> RegionFormatChanged;
+
+        /// <summary>
+        /// Occurs when the device orientation is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<DeviceOrientationEventArgs> DeviceOrientationChanged;
+
+        /// <summary>
+        /// Occurs when the device orientation is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<SuspendedStateEventArgs> SuspendedStateChanged;
+
+        /// <summary>
+        /// A component instance ID.
+        /// It will be created after OnCreate method is invoked.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public string Id { get; internal set; }
+
+        /// <summary>
+        /// A component ID
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public string ComponentId { get; internal set; }
+
+        /// <summary>
+        /// Parent object
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public CBApplicationBase Parent { get; internal set; }
+
+        /// <summary>
+        /// Finish current component
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Finish()
+        {
+            Interop.CBApplication.ComponentFinish(Handle);
+        }
+
+        internal void Bind(IntPtr handle, string compId, CBApplicationBase parent)
+        {
+            Handle = handle;
+            ComponentId = compId;
+            Parent = parent;
+        }
+
+        /// <summary>
+        /// Overrides this method if you want to specify a type of this component.
+        /// Default component type is Service type.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public virtual ComponentType GetComponentType()
```

Comment:
Why don't you use Property?
```suggestion
        public virtual ComponentType ComponentType { get; }
```

Reply:
Done